util/stream.go

```diff
@@ -32,12 +30,36 @@ type Parser interface {
 	Parse(b []byte) (message Message, err error)
 }
 
+type streamWorker struct {
+	Full chan *bytes.Buffer
+}
+
+func (w *streamWorker) parse(stopCh chan bool, parser Parser, inbound chan Message, empty chan *bytes.Buffer) {
+	errMessage := "received: %v and encountered error: %v"
+	for {
+		select {
+		case b := <-w.Full:
+			msg, err := parser.Parse(b.Bytes())
+			// Log all message parsing errors.
+			if err != nil {
+				log.Errorf(errMessage, b.Bytes(), err)
+			}
```

Comment:
I know the original code works this way, but does it make sense to send `msg` on the `inbound` channel in the presence of errors (`inbound <- msg`)?

Reply:
Updated.

Reply:
you are leaking buffers now

either we should
1) keep the previous version if you think `inbound <- msg` is fine even in case of errors
2) rewrite the code like this
```go
			if err != nil {
				log.Errorf(errMessage, b.Bytes(), err)
			} else {
				inbound <- msg
			}
			b.Reset()
			empty <- b
```

Reply:
Updated.